mmeval/metrics/rouge.py

```diff
@@ -0,0 +1,327 @@
+            if isinstance(predictions, str):
+                references = [references]
+            else:
+                if len(predictions) == 1:
+                    references = [references]
+                else:
+                    references = [[reference] for reference in references]
+
+        if isinstance(predictions, str):
+            predictions = [predictions]
+
+        if isinstance(references, str):
+            references = [[references]]
+        assert len(predictions) == len(
+            references
+        ), 'The number of predictions and references must be equal'
+
+        # If the tokenizer is None, check the first sentence
+        # to determine which language the tokenizer is used.
+        if self.tokenizer_fn is None:
+            lang = 'en'
+            for _char in predictions[0]:
+                if '\u4e00' <= _char <= '\u9fa5':
+                    lang = 'cn'
+                    break
+            self.tokenizer_fn = _get_tokenizer(lang)
+        # Traverse the predicted sentences
+        for prediction, _references in zip(predictions, references):
+            pred_token = _normalize_and_tokenize(prediction, self.stemmer,
+                                                 self.normalizer,
+                                                 self.tokenizer_fn)
+            reference_len = len(_references)
+            ref_tokens = []
+            results = []
+            for reference in _references:
+                ref_token = _normalize_and_tokenize(reference, self.stemmer,
+                                                    self.normalizer,
+                                                    self.tokenizer_fn)
+                ref_tokens.append(ref_token)
+            # Traverse the chosen rouge_keys
+            for i, rouge_key in enumerate(self.rouge_keys):
+                fmeasure = precision = recall = 0.
+                # Traverse the tokens of references for single prediction.
+                for j, ref_token in enumerate(ref_tokens):
+                    if isinstance(rouge_key, int):
+                        score = _rougeN_score(pred_token, ref_token, rouge_key)
+                    else:
+                        score = _rougeL_score(pred_token, ref_token)
+                    if self.accumulate == 'best':
+                        if fmeasure < score['fmeasure']:
+                            fmeasure = score['fmeasure']
+                            precision = score['precision']
+                            recall = score['recall']
+                    else:
+                        fmeasure += score['fmeasure']
+                        precision += score['precision']
+                        recall += score['recall']
+                if self.accumulate == 'avg':
+                    fmeasure, precision, recall = \
+                        fmeasure / reference_len, \
+                        precision / reference_len, recall / reference_len
```

Comment:
Suggestion that improve code readability
```suggestion
                    fmeasure = fmeasure / reference_len
                    precision = precision / reference_len
                    recall = recall / reference_len
```